src/frontends/tensorflow_common/src/op/adjust_saturation.cpp

```diff
@@ -0,0 +1,188 @@
+
+    return make_shared<tuple<shared_ptr<Node>, shared_ptr<Node>, shared_ptr<Node>>>(hh_final, ss, vv);
+    // return make_shared<v0::Concat>(NodeVector{hh, vv, s}, -1); 
+}
+
+shared_ptr<Node> hsv_to_rgb(shared_ptr<Node> h, shared_ptr<Node> s, shared_ptr<Node> v, element::Type type) {
+    // c = s * v;
+    auto c = make_shared<v1::Multiply>(s, v);
+    // m = v - c;
+    auto m = make_shared<v1::Subtract>(v, c);
+    // dh = h * 6;
+    auto dh = make_shared<v1::Multiply>(h, make_shared<v0::Constant>(type, Shape{}, 6.0f));
+
+    // fmodu rounded to within [0, 2)
+    auto fmodu = make_shared<v1::FloorMod>(dh, make_shared<v0::Constant>(type, Shape{}, 2.0f));
+
+    //  x = c * (1 - std::abs(fmodu - 1));
+    auto x = make_shared<v1::Multiply>(c, make_shared<v1::Subtract>(make_shared<v0::Constant>(element::f32, Shape{}, 1.0f), make_shared<v0::Abs>(make_shared<v1::Subtract>(fmodu, make_shared<v0::Constant>(type, Shape{}, 1.0f)))));
+
+    
+    // h_category = static_cast<int>(dh);
+    auto h_category = make_shared<v0::Convert>(make_shared<v0::Floor>(dh), element::i32);
+
+    auto zeros = make_shared<v0::Constant>(type, x->get_shape(), 0.0f);
+
+    auto rr_options = NodeVector{c, x, zeros, zeros, x, c};
+    auto gg_options = NodeVector{x, c, c, x, zeros, zeros};
+    auto bb_options = NodeVector{zeros, zeros, x, c, c, x};
+
+    auto rr_concat = make_shared<v0::Concat>(rr_options, -1); 
+    auto gg_concat = make_shared<v0::Concat>(gg_options, -1);
+    auto bb_concat = make_shared<v0::Concat>(bb_options, -1);
+    
+    // Use a gather operation to select the correct channel values based on h_category
+    int batch_dim = s->get_shape().size() - 1;
+
+    auto axis = make_shared<v0::Constant>(element::i32, Shape{}, -1);
+    auto rr = make_shared<v8::Gather>(rr_concat, h_category, axis, batch_dim);
+    auto gg = make_shared<v8::Gather>(gg_concat, h_category, axis, batch_dim);
+    auto bb = make_shared<v8::Gather>(bb_concat, h_category, axis, batch_dim);
+
+    // Adding m to each component
+    auto r = make_shared<v1::Add>(rr, m);
+    auto g = make_shared<v1::Add>(gg, m);
+    auto b = make_shared<v1::Add>(bb, m);
+
+    // Return concatenated RGB 
+    return make_shared<v0::Concat>(NodeVector{r, g, b}, -1); 
+}
+
+OutputVector translate_adjust_saturation_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"AdjustSaturation"});
+    auto images = node.get_input(0);
+    auto scale = node.get_input(1);
+    auto node_name = node.get_name();
+
+    auto type = images.get_element_type();
+
+    auto hsv_components = convert_rgb_to_hsv(images.get_node_shared_ptr(), type);
+    auto hh = get<0>(*hsv_components);
+    auto ss = get<1>(*hsv_components);
+    auto vv = get<2>(*hsv_components);
+
+    scale = make_shared<v1::ConvertLike>(scale, images);
+
+    auto ss_adjust = make_shared<v0::Clamp>(make_shared<v1::Multiply>(ss, scale), 0.0f, 1.0f);
```

Comment:
is it really required? Though, I understand it can be important when we converting from one color space to another. Please confirm. Thanks

Reply:
https://github.com/tensorflow/tensorflow/blob/e1164b533e9eaa337035f73e3f10ba1f6db430ed/tensorflow/core/kernels/image/adjust_saturation_op.cc#L206

To be consistent with tf implementation. Missing this part took me a long time to debug.